src/process-mining-visualisation/mining-action.ts

```diff
@@ -57,7 +44,15 @@ export class MiningCommand extends Command {
 
   execute(context: CommandExecutionContext): CommandReturn {
     const model = context.root;
-    this.action.data.nodes.forEach(node => {
+    if (model.children.filter(e => e.type === DiagramCaption.TYPE).length > 0) {
+      return model;
+    }
+    return this.populate(model);
+  }
+
+  async populate(model: SModelRootImpl) {
+    const data = await (await fetch(this.miningData.url)).json();
```

Comment:
json() returns an any type if I'm not wrong. 
So I would recommend that you don't delete the types above but cast it here so you're editor can help you to not write typos...